temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java

```diff
@@ -410,6 +426,47 @@ private Worker createNewWorker(
     return worker;
   }
 
+  private void addRegisteredWorkflowImpl(Worker worker, String workflowClass) {
+    if (!registeredInfo.containsKey(worker.getTaskQueue())) {
+      registeredInfo.put(
+          worker.getTaskQueue(), new RegisteredInfo().addWorkflowInfo(workflowClass));
+    } else {
+      registeredInfo.get(worker.getTaskQueue()).getRegisteredWorkflowInfo().add(workflowClass);
+    }
+  }
+
+  private void addRegisteredActivityImpl(Worker worker, String activityClass) {
+    if (!registeredInfo.containsKey(worker.getTaskQueue())) {
+      registeredInfo.put(
+          worker.getTaskQueue(), new RegisteredInfo().addActivityInfo(activityClass));
+    } else {
+      registeredInfo.get(worker.getTaskQueue()).getRegisteredActivityInfo().add(activityClass);
+    }
+  }
+
+  public static class RegisteredInfo {
+    private final List<String> registeredActivityInfo = new ArrayList<>();
+    private final List<String> registeredWorkflowInfo = new ArrayList<>();
+
+    public RegisteredInfo addActivityInfo(String activityClass) {
+      registeredActivityInfo.add(activityClass);
+      return this;
+    }
+
+    public RegisteredInfo addWorkflowInfo(String workflowClass) {
+      registeredWorkflowInfo.add(workflowClass);
+      return this;
+    }
+
+    public List<String> getRegisteredActivityInfo() {
```

Comment:
Do you think there is other information users may want us to expose in the future about activities or workflows other then their name?

Reply:
May make sense to return a list of objects so we can add new fields if decided to expose more info.

Reply:
ok good point, will update